### ISSUE TITLE
How to save a sparkxgbregressor model to local?

### ISSUE THREAD

**Issue by TurtlelovePig** [dmlc/xgboost#11361]
I have trained a sparkxgbregression model on spark, and can save it use "model.save()"; however, i want to save this model to local so it can be read and processed throught cpp; i use the following way to save the model: "booster = model.get_booster(); booster.save_model("xxx.json")".
but the problem is that when i load the local json model using python or cpp, the prediction is inconsistent with the sparkmodel, i am sure i am using the same data and i wonder where is the key?

**Comment by trivialfis** [comment]
Could you please provide some details about your model? Is it a classification model? If so, then the spark package might do some post-processing like obtaining the softmax prediction, etc.

**Comment by TurtlelovePig** [comment]
> Could you please provide some details about your model? Is it a classification model? If so, then the spark package might do some post-processing like obtaining the softmax prediction, etc.

sure! the model is a regression model, and all feature is used as "q"

**Comment by trivialfis** [comment]
Thank you for the model info. I will try to reproduce it, but if I can't, I will ask for some example scripts and data from you.

**Comment by TurtlelovePig** [comment]
> Thank you for the model info. I will try to reproduce it, but if I can't, I will ask for some example scripts and data from you.

thank u; I am glad to provide those if u need

**Comment by trivialfis** [comment]
> thank u; I am glad to provide those if u need

Please share it if it's available. ;-)

**Comment by TurtlelovePig** [comment]
> > thank u; I am glad to provide those if u need
> 
> Please share it if it's available. ;-)

I guess i found where the problem is; when i train this model on spark, i use the "callback function" to use a decay learning rate; but when i save the json local model, the decay learning rate seems not work? so there is difference? I have proved this on a small dataset, i will try it on a larger one

**Comment by TurtlelovePig** [comment]
> > > thank u; I am glad to provide those if u need
> > 
> > 
> > Please share it if it's available. ;-)
> 
> I guess i found where the problem is; when i train this model on spark, i use the "callback function" to use a decay learning rate; but when i save the json local model, the decay learning rate seems not work? so there is difference? I have proved this on a small dataset, i will try it on a larger one

sorry this seems to be wrong; i am still working on it

**Comment by TurtlelovePig** [comment]
@trivialfis After thorough investigation, the issue was ultimately identified as being caused by discrepancies in missing value handling between Spark and the local environment. Issue closed.

**Comment by TurtlelovePig** [comment]
@trivialfis I have turned some params like "missing" to right, but there is still huge difference between json version's prediction and sparks'; so i wonder if there are some other params like missing will lead to this issue?

**Comment by TurtlelovePig** [comment]
> Hi [@TurtlelovePig](https://github.com/TurtlelovePig)
> 
> > thank u; I am glad to provide those if u need
> 
> Would you like to share the example data and script? I can help debug it.

yes, i can upload a dataset about 30g, a spark model file and a corresponding json model file, is this useful? or do u need something else ?

**Comment by trivialfis** [comment]
Would a subset of the data be sufficient to reproduce the issue? Please make sure the dataset and the script don't contain any sensitive information. Also, it would be great if I could run the script directly and see the error.

**Comment by TurtlelovePig** [comment]
> Would a subset of the data be sufficient to reproduce the issue? Please make sure the dataset and the script don't contain any sensitive information. Also, it would be great if I could run the script directly and see the error.

The issue is caused because i do not set the missing value when create dmatirx; thank you!!!
